packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -173,7 +173,7 @@ bool isOlderThanReference({ @required FileSystemEntity entity, @required File re
     return true;
   }
   return referenceFile.existsSync()
-      && referenceFile.lastModifiedSync().isAfter(entity.statSync().modified);
+      && referenceFile.statSync().modified.isAfter(entity.statSync().modified);
```

Comment:
These 2 different APIs have different granularity so if entity is modified within a second of referenceFile was, the result will be wrong and all kinds of crazy things can happen.